dsgrid/dimension/dimension_filters.py

```diff
@@ -0,0 +1,230 @@
+import abc
+import logging
+from typing import Any, Dict, Union
+
+import pyspark.sql.functions as F
+from pydantic import Field, validator, root_validator
+
+from dsgrid.data_models import DSGBaseModel
+from dsgrid.dimension.base_models import DimensionType
+
+
+logger = logging.getLogger(__name__)
+
+
+class DimensionFilterBaseModel(DSGBaseModel, abc.ABC):
+    """Base model for all filters"""
+
+    dimension_type: DimensionType
+    dimension_query_name: str
+    column: str = Field(
+        title="column", description="Column of dimension records to use", default="id"
+    )
+
+    @abc.abstractmethod
+    def apply_filter(self, df):
+        """Apply the filter to a DataFrame"""
+
+    @staticmethod
+    def _make_value_str(value):
+        if isinstance(value, str):
+            return f"'{value}'"
+        return str(value)
+
+    @staticmethod
+    def _make_values_str(values):
+        if isinstance(values[0], str):
+            text = ", ".join((f"'{x}'" for x in values))
+        elif isinstance(values[0], int) or isinstance(values[0], float):
+            text = ", ".join((f"{x}" for x in values))
+        else:
+            raise Exception(f"Unsupported type: {type(values[0])}")  # TODO: dsg exception
```

Comment:
Make this func consistent with `_make_value_str(value)` above? If you have unsupported type here, I'd imagine it's the same above. Better if `_make_values_str(values)` = a func of `_make_value_str(value)`